### ISSUE TITLE
Logtalk 3.56 application installation issue.

### ISSUE THREAD

**Issue by deviAshokkumar** [LogtalkDotOrg/logtalk3#148]
I am validating the LogTalk3.5.6.exe application silently , but however I can see below error message during silent installation for the first time.

 

Parameters are used below but still I can see the error.

"<Path of logtalk-3.56.0.exe>" /SP- /VERYSILENT /SUPPRESSMSGBOX /NORESTART 

Also however for uninstallation , it is going to user documents folder where as installed with admin Or upon installing with system context it is installing in system profile.

Could you please help me to provide install and uninstallation instructions with command line for silent, that would be helpful to me to validate.

Regards,
Ashok

**Comment by pmoura** [comment]
You didn't include the error message... From your mail yesterday:

> Failed to expand shell folder constant "userdocs"

Doing some search in this error, it seems the issue may be due to a Windows setup where the "My Documents" folder either doesn't exist or is relocated to a network drive. Can you confirm?

It would also help to know which Windows version are you using and under which user is the installer being run.

**Comment by pmoura** [comment]
Does your admin user have a local "My Documents" folder?

**Comment by deviAshokkumar** [comment]
Yes, Of course. My understanding is while extracting the files during install, it is giving below error in log file.

Log file:

2022-07-08 15:13:52.198   Log opened. (Time zone: UTC+05:30)
2022-07-08 15:13:52.198   Setup version: Inno Setup version 6.2.0
2022-07-08 15:13:52.198   Original Setup EXE: C:\temp2\logtalk-3.56.0.exe
2022-07-08 15:13:52.198   Setup command line: /SL5="$40382,10536813,831488,C:\temp2\logtalk-3.56.0.exe" /SP- /VERYSILENT /SUPPRESSMSGBOXES /NORESTART /LOG="C:\Temp\Install1.log"
2022-07-08 15:13:52.198   Windows version: 10.0.19044  (NT platform: Yes)
2022-07-08 15:13:52.198   64-bit Windows: Yes
2022-07-08 15:13:52.198   Processor architecture: x64
2022-07-08 15:13:52.198   User privileges: Administrative
2022-07-08 15:13:52.245   Administrative install mode: Yes
2022-07-08 15:13:52.245   Install mode root key: HKEY_LOCAL_MACHINE
2022-07-08 15:13:52.261   64-bit install mode: No
2022-07-08 15:13:52.276   Created temporary directory: C:\Windows\TEMP\is-D4LJ9.tmp
2022-07-08 15:13:53.104   Warning: SHGetFolderPath failed with code 0x80070002 on folder 0x8005
2022-07-08 15:13:53.104   InitializeWizard raised an exception (fatal).
2022-07-08 15:13:53.104   Exception message:
2022-07-08 15:13:53.104   Defaulting to OK for suppressed message box (OK):
                          Runtime error (at 48:2857):
                          
                          Internal error: Failed to expand shell folder constant "userdocs"
2022-07-08 15:13:53.104   Deinitializing Setup.
2022-07-08 15:13:53.104   Log closed.

**Comment by pmoura** [comment]
Assuming my understanding of system context is correct (I'm not a regular Windows user), the installer cannot perform the default full installation when run in this context, hence the error. Try using `/COMPONENTS=base`.

**Comment by deviAshokkumar** [comment]
You mean try with :           /COMPONENTS="base" /SP- /VERYSILENT /SUPPRESSMSGBOXES /NORESTART /LOG="C:\Temp\Install1.log" with full command line?

**Comment by deviAshokkumar** [comment]
It seems to be not working, below is log file.

2022-07-08 15:28:47.488   Log opened. (Time zone: UTC+05:30)
2022-07-08 15:28:47.488   Setup version: Inno Setup version 6.2.0
2022-07-08 15:28:47.488   Original Setup EXE: C:\temp2\logtalk-3.56.0.exe
2022-07-08 15:28:47.488   Setup command line: /SL5="$305FA,10536813,831488,C:\temp2\logtalk-3.56.0.exe" /COMPONENTS="base" /SP- /VERYSILENT /SUPPRESSMSGBOXES /NORESTART /LOG="C:\Temp\Install1.log"
2022-07-08 15:28:47.488   Windows version: 10.0.19044  (NT platform: Yes)
2022-07-08 15:28:47.488   64-bit Windows: Yes
2022-07-08 15:28:47.488   Processor architecture: x64
2022-07-08 15:28:47.488   User privileges: Administrative
2022-07-08 15:28:47.535   Administrative install mode: Yes
2022-07-08 15:28:47.535   Install mode root key: HKEY_LOCAL_MACHINE
2022-07-08 15:28:47.535   64-bit install mode: No
2022-07-08 15:28:47.535   Created temporary directory: C:\Windows\TEMP\is-3IT1H.tmp
2022-07-08 15:28:48.316   Warning: SHGetFolderPath failed with code 0x80070002 on folder 0x8005
2022-07-08 15:28:48.316   InitializeWizard raised an exception (fatal).
2022-07-08 15:28:48.316   Exception message:
2022-07-08 15:28:48.316   Defaulting to OK for suppressed message box (OK):
                          Runtime error (at 48:2857):
                          
                          Internal error: Failed to expand shell folder constant "userdocs"
2022-07-08 15:28:48.316   Deinitializing Setup.
2022-07-08 15:28:48.332   Log closed.

**Comment by pmoura** [comment]
It seems then that running the installer in system context is not possible without changes to the installer itself. The installer currently requires to be run by the final user (which can be an admin user).

**Comment by deviAshokkumar** [comment]
But we would like to test the app using system context, since we may push the app to the end users.  Any alternative solution?

**Comment by pmoura** [comment]
Not that I'm aware. For now, you will need to validate the installer without using system context, running it with either an admin user or a non-privileged user. But I'm open to pull request with fixes/improvements to the [installer](https://github.com/LogtalkDotOrg/logtalk3/tree/master/scripts/windows).

**Comment by pmoura** [comment]
Fixed in db79d406b5a659d007a13665c832ea65839c9aba.
